### PR TITLE
feat: dtm verify logic and logging improvement

### LINKED ISSUE #252: `dtm verify` results show that some tools are not healthy, but can't get more detail
## Describe the Bug

dtm version: dtm-darwin-arm64 0.2.0 
OS: MacOS  M1
config file: 
```yaml
---
tools:
- name: golang-demo-repo
  plugin:
    kind: github-repo-scaffolding-golang
    version: 0.2.0
  options:
    owner: ironcore864
    # will also be used as application name
    repo: golang-demo
    branch: main
    image_repo: ironcore864/golang-demo
- name: golang-demo-app
  plugin:
    kind: githubactions-golang
    version: 0.2.0
  options:
    owner: ironcore864
    repo: golang-demo
    language:
      name: go
      version: "1.17"
    branch: main
    build:
      enable: True
    test:
      enable: True
      coverage:
        enable: True
    docker:
      enable: True
- name: go-hello-world-gitlab-ci
  plugin:
    kind: gitlabci-golang
    version: 0.2.0
  options:
    pathWithNamespace: ironcore864/golang-demo
    branch: main
- name: devlake
  plugin:
    kind: devlake
    version: 0.2.0
```

## How to Reproduce

```bash
$ dtm init
$ dtm verify
```

## Screenshots

image

### LINKED ISSUE #255: Bug: `dtm verify` shouldn't return any error before `dtm apply`
## Describe the Bug

If the users forget to execute `dtm apply` after `dtm init`, but directly execute `dtm verify`, error log should not be output at this time.

See the https://github.com/merico-dev/stream/issues/252#issue-1156718149 for more info.

## How to Reproduce

See the https://github.com/merico-dev/stream/issues/252#issue-1156718149 for more info.

### PR DESCRIPTION
**Pull request by IronCore864**
# Summary

1. dtm verify should not "fail" or "error" in any circumstances. The purpose of the "verify", as the word means, is to make sure if something is as expected.
2. The "error" in the log is a bit too much. If an "error" message is displayed, the user might think something went wrong or something they did is wrong, which isn't the case in the original post's issue.
3. we should have dedicated documentation explaining how everything works.

## Key Points

- [x] This is a breaking change.
- [x] Documentation (new or existing) is updated.

### Related Issues

Closes #255

Closes #252 

## Test Results

```bash
$ ./dtm verify
2022-03-03 11:03:15 ℹ [INFO]  Verify started.
2022-03-03 11:03:15 ℹ [INFO]  Using dir <.devstream> to store plugins.
2022-03-03 11:03:15 ℹ [INFO]  Tool < golang-demo-repo > found in config but doesn't exist in the state, will be created.
2022-03-03 11:03:15 ℹ [INFO]  Tool < golang-demo-app > found in config but doesn't exist in the state, will be created.
2022-03-03 11:03:15 ℹ [INFO]  Verify finished.
$ rm -rf .devstream/
$ ./dtm verify
2022-03-03 11:03:27 ℹ [INFO]  Verify started.
2022-03-03 11:03:27 ℹ [INFO]  Using dir <.devstream> to store plugins.
2022-03-03 11:03:27 ℹ [INFO]  stat .devstream/github-repo-scaffolding-golang-darwin-arm64_0.2.0.so: no such file or directory
2022-03-03 11:03:27 ℹ [INFO]  Maybe you forgot to run "dtm init" first?
2022-03-03 11:03:27 ℹ [INFO]  Verify finished.
```

### CODE REVIEW

**Comment by IronCore864** [comment]
@imxw have a look if you are interested, and more feedback is warmly welcome!

File: docs/commands_in_detail/verify.md [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+
+The command `dtm verify` checks the following:
+
+## 1. Config File
+
+`dtm verify` first verifies if the config file can be loaded successfully.
+
+If not, the following information might be printed out:
+
+- if the config file doesn't exist, it reminds you if you forgot to specify the config file by using the "-f" parameter;
```
Comment (daniel-hutao) [inline docs/commands_in_detail/verify.md:11]: If ?

File: internal/pkg/pluginengine/change_helper.go [hunk 2]
```diff
@@ -84,17 +86,16 @@ func changesForApply(smgr statemanager.Manager, cfg *configloader.Config) ([]*Ch
...
-					log.Infof("Change added: %s -> %s", tool.Name, statemanager.ActionCreate)
+					// tool exists in the state, but resource doesn't exist, Create
+					description := fmt.Sprintf("Tool < %s > state found but it seems the tool isn't created, will be created.", tool.Name)
+					changes = append(changes, generateCreateAction(&tool, description))
 				} else if drifted(resource, state.Resource) {
-					log.Debugf("Tool %s %s resource drifted from state.", tool.Name, tool.Plugin.Kind)
 					// resource drifted from state, Update
-					changes = append(changes, generateUpdateAction(&tool))
-					log.Infof("Change added: %s -> %s", tool.Name, statemanager.ActionUpdate)
+					description := fmt.Sprintf("Tool < %s > drifted from the state, will be updated.", tool.Name)
```
Comment (lfbdev) [inline internal/pkg/pluginengine/change_helper.go:94]: Perhaps the 'state' information is invisible to the user?

**Review by lfbdev — APPROVED** [review]

File: docs/commands_in_detail/verify.md (hunk 1 above)
Comment (IronCore864) [reply docs/commands_in_detail/verify.md:11]: ## Some common punctuation usages for "bullet points":

### Main Sentence with Sentence Fragments

Main sentence blabla:
- sentence fragment 1 without punctuation
- sentence fragment 1 without punctuation
- sentence fragment 3 without punctuation

### Main Sentence with Clauses

Main sentence blabla:
- clause 1 not ending with a period;
- clause 2 not ending with a period;
- clause 3 not ending with a period.

### Main Sentence with Full Sentences

Main sentence blabla:
- Full-sentence 1 ending with a period.
- Full-sentence 2 ending with a period.
- Full-sentence 3 ending with a period.

See [this post](https://www.linkedin.com/pulse/how-capitalize-punctuate-bullet-points-l-michelle-baker-phd/) for more information: 

I recommend [Dreyer's English](https://www.amazon.com/Dreyers-English-Utterly-Correct-Clarity/dp/0812995708) if you are interested in English writing :) I find it helpful and humourous.

File: internal/pkg/pluginengine/change_helper.go (hunk 2 above)
Comment (IronCore864) [reply internal/pkg/pluginengine/change_helper.go:94]: Very good point. I don't have a better description here; the main purpose is to let the user know why the "Upgrade" change is required. If we put only "Tool xxx will be updated", the user might think: why?

Is there a better way to put it? 
@lfbdev @daniel-hutao

File: docs/commands_in_detail/verify.md (hunk 1 above)
Comment (daniel-hutao) [reply docs/commands_in_detail/verify.md:11]: 👍👍👍

**Review by daniel-hutao — APPROVED** [review]

File: internal/pkg/pluginengine/change_helper.go (hunk 2 above)
Comment (daniel-hutao) [reply internal/pkg/pluginengine/change_helper.go:94]: I  agree leave the state here for now.

**Review by daniel-hutao — APPROVED** [review]

**Pull request merged**